### PR TITLE
Methods For StartEpoch and Slashing Helper Types for Optimized Slasher

### PR DESCRIPTION
**Pull request by rauljordan**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**

When performing slashing surround detection, we need to figure out the start epoch of a chunk we are updating. This differs depending on whether the chunk is a min span chunk or max span chunk. As such, we should add methods for this as part of the `Chunker` interface. Additionally, this PR adds a useful `Slashing` type which summarizes all the information about surround vote slashings.

**Which issues(s) does this PR fix?**

Part of #8331

### CODE REVIEW

File: beacon-chain/slasher/chunks.go [hunk 1]
```diff
@@ -375,10 +389,89 @@ func (m *MaxSpanChunksSlice) Update(
...
+//  max_spans_val_i = [[0, 0, 0], [0, 0, 0], [0, 0, 0]]
+//
+// If C = ChunkSize is 3 epochs per chunk, and we input start epoch of chunk 1 which is 3. The next start
+// epoch is the start epoch of chunk 2, which is epoch 6. This is computed as:
+//
+//  (start_epoch / C) + 1 * C
+//  (3 / 3) + 1 * 3 = 6
+//
+func (m *MaxSpanChunksSlice) NextChunkStartEpoch(startEpoch types.Epoch) types.Epoch {
+	return types.Epoch((uint64(startEpoch)/m.params.chunkSize + 1) * m.params.chunkSize)
```
Comment (prestonvanloon) [inline beacon-chain/slasher/chunks.go:472]:
```suggestion
	return types.Epoch(((uint64(startEpoch)/m.params.chunkSize) + 1) * m.params.chunkSize)
```

Add params to match the examples in the godoc. Doesn't change the logic but maybe helps align reader with the godoc.

File: beacon-chain/slasher/chunks.go [hunk 2]
```diff
@@ -375,10 +389,89 @@ func (m *MaxSpanChunksSlice) Update(
...
+//  max_spans_val_i = [[0, 0, 0], [0, 0, 0], [0, 0, 0]]
+//
+// If C = ChunkSize is 3 epochs per chunk, and we input start epoch of chunk 1 which is 3. The next start
+// epoch is the last epoch of chunk 0, which is epoch 2. This is computed as:
+//
+//  (start_epoch / C) * (C - 1)
+//  (3 / 3) * (3 - 1) = 1 * 2 = 2
+//
+func (m *MinSpanChunksSlice) NextChunkStartEpoch(startEpoch types.Epoch) types.Epoch {
+	return types.Epoch(uint64(startEpoch)/m.params.chunkSize*m.params.chunkSize - 1)
```
Comment (prestonvanloon) [inline beacon-chain/slasher/chunks.go:455]:
```suggestion
	return types.Epoch((uint64(startEpoch)/m.params.chunkSize)*(m.params.chunkSize - 1))
```
Add parenthesis to match godoc description.

This actually changes the logic. Please double check that the godoc and implementation match and are correct.

**Review by prestonvanloon — CHANGES_REQUESTED** [review]

File: beacon-chain/slasher/params.go [hunk 3]
```diff
@@ -63,7 +92,7 @@ func (p *Parameters) validatorChunkIndex(validatorIndex types.ValidatorIndex) ui
 //     val0     val1     val2
 //      |        |        |
 //   {     }  {     }  {     }
-//  [2, 2, 2, 2, 2, 2, 2, 2, 2]
```
Comment (rauljordan) [inline beacon-chain/slasher/params.go:66]: The contents of the slice are not relevant here, so removed to reduce confusion

**Review by prestonvanloon — APPROVED** [review]
Looks great!

**Pull request merged**